File: 34. Find First and Last Position of Element in Sorted Array/Solution.py
class Solution:
    def searchRange(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        i, j = 0, len(nums) - 1
        while i <= j:
            mid = (i + j) >> 1
            if nums[mid] == target:
                left, right = mid, mid
                while left >= 0 and nums[left] == target:
                    left -= 1
                while right <= len(nums) - 1 and nums[right] == target:
                    right += 1
                return [left + 1, right - 1]
            elif nums[mid] < target:
                i = mid + 1
            else:
                j = mid - 1
        return [-1, -1]

File: 34. Find First and Last Position of Element in Sorted Array/Solution.cpp
class Solution
{
  public:
    vector<int> searchRange(vector<int> &nums, int target)
    {
        int i = 0, j = nums.size() - 1;
        while (i <= j)
        {
            int mid = (i + j) >> 1;
            if (nums[mid] == target)
            {
                int left = mid, right = mid;
                while (left >= 0 && nums[left] == target)
                    left--;
                while (right <= nums.size() - 1 && nums[right] == target)
                    right++;
                return {left + 1, right - 1};
            }
            else if (nums[mid] < target)
            {
                i = mid + 1;
            }
            else
            {
                j = mid - 1;
            }
        }
        return {-1, -1};
    }
};